### PR TITLE
View compatible target in Target MVD

### PR DESCRIPTION
**Pull request by afmicc**
#### Trello board reference:

* [As a user I should be able to see all my compatible targets so I can start a conversation with any of my compatible users](https://trello.com/c/gD8Olo9W)
* [As a user I should be able to see all my conversations](https://trello.com/c/EtlFWlJl)

---

#### Description:

* Change method to search targets: `Target.near`, because it doesn't work as I expected
* Add get compatible targets action
* Add request spec test to validate compatible targets functionality

### CODE REVIEW

File: spec/request/api/v1/targets/compatible_spec.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        get compatible_api_v1_targets_path, headers: auth_header
+      end
+
+      it 'is expected a successful response' do
+        expect(response).to be_successful
+      end
+
+      it 'is expected that response contains at least one' do
+        body = JSON response.body
+        expect(body).to_not be_empty
```
Comment (gabrielbursztein2) [inline spec/request/api/v1/targets/compatible_spec.rb]: You should test that `body[:targets]` it not empty instead.

File: spec/request/api/v1/targets/compatible_spec.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  end
+
+  context 'when there are multi target matchs' do
+    include_context 'near targets'
+
+    let!(:target) do
+      attributes = target_new[:target]
+      attributes[:user] = user
+      create(:target, attributes)
+    end
```
Comment (gabrielbursztein2) [inline spec/request/api/v1/targets/compatible_spec.rb]: `let!(:target) { create(:target, user: user)`

File: spec/request/api/v1/targets/compatible_spec.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    it 'is expected a successful response' do
+      expect(response).to be_successful
+    end
+
+    it 'is expected that response contains 2 targets' do
+      body = JSON response.body
+      targets = json_value(body, 'targets')
+      expect(targets).to_not be_empty
```
Comment (gabrielbursztein2) [inline spec/request/api/v1/targets/compatible_spec.rb]: Remove this line, it's unnecessary.

**Review by gabrielbursztein2 — APPROVED** [review]
Good work!

File: config/routes.rb [hunk 4]
```diff
@@ -7,7 +7,9 @@
     namespace :v1, defaults: { format: :json } do
       devise_scope :user do
         resources :users, only: :update
-        resources :targets, only: %i[index create destroy]
+        resources :targets, only: %i[index create destroy] do
+          get 'compatible', on: :collection
```
Comment (JuanfraM) [inline config/routes.rb:11]: I like this approach: 
```
collection do
  get :compatible
end
```
but it's up to u.

**Review by JuanfraM — APPROVED** [review]
nice job 💪

**Pull request merged**